nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -470,6 +477,8 @@ def _check_param_groups_mismatch(self, checkpoint_path: Union[str, Path], sharde
         # @akoumparouli: check if it contains an mcore dist opt
         if common_state_dict.get('optimizer_states', [{}])[0].get('param_groups', None) is None:
             return False
+        if sharded_state_dict.get('optimizer_states') is None:
```

Comment:
[Minor] If you put it before the previous check, you can simplify the `common_state_dict.get('optimizer_states', [{}])` to `common_state_dict['optimizer_states']`